### PR TITLE
Add support for iscsi vlan tagging

### PR DESCRIPTION
**Pull request by shaik-gousa-khaja-asif**
Referenced the PR https://github.com/hpe-storage/python-3parclient/pull/30 and made changes on 
"from hpe3parclient import showport_parser" and "return {'iSCSIPortInfo' : {'vlan' : vlan}}"

### CODE REVIEW

File: hpe3parclient/client.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
 import re
 import time
 import uuid
+import copy
```
Comment (sneharai4) [inline hpe3parclient/client.py]: Please arrange imports in alphabetical order.

File: hpe3parclient/showport_parser.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+ports with vlan tags.
+
+"""
+
+
+class ShowportParser:
+    """ Parses the following showport commands on an HP 3par array
+        showport
+        showport -iscsi
+        showpoer -iscsivlan
```
Comment (sneharai4) [inline hpe3parclient/showport_parser.py]: Typo error: showport -iscsivlan

File: hpe3parclient/client.py [hunk 3]
```diff
@@ -1895,6 +1912,36 @@ def _getProtocolPorts(self, protocol, state=None):
...
+                new_port = copy.deepcopy(matching_ports[0])
+                new_port.update(port)
+                cloned_ports.append(new_port)
+
+        return cloned_ports
+
+    def _getISCSIVlans(self):
+        self.ssh.open()
+        iscsi_vlan_data = self.ssh.run(['showport', '-iscsivlans'])
+        self.ssh.close()
```
Comment (sneharai4) [inline hpe3parclient/client.py]: Instead of this, we can directly do 
return self._run(['showport', '-iscsivlans'])

**Comment by sneharai4** [comment]
@shaik-gousa-khaja-asif Please check https://github.com/hpe-storage/python-3parclient/blob/master/hpe3parclient/client.py#L3748  as it has return body. But it will throw error for variable not defined, if removeFromTarget is set to true.

**Comment by sneharai4** [comment]
@shaik-gousa-khaja-asif  Also please add unittests.

**Comment by wdurairaj** [comment]
> @shaik-gousa-khaja-asif Also please add unittests.

Unit tests are there in test/test_HPE3ParClient_ShowportParser.py

File: hpe3parclient/client.py [hunk 4]
```diff
@@ -1895,6 +1920,36 @@ def _getProtocolPorts(self, protocol, state=None):
...
+                    NoUniqueMatch(err.format(len(matching_ports), port))
+
+            if len(matching_ports) == 1:
+                new_port = copy.deepcopy(matching_ports[0])
+                new_port.update(port)
+                cloned_ports.append(new_port)
+
+        return cloned_ports
+
+    # def _getISCSIVlans(self):
```
Comment (wdurairaj) [inline hpe3parclient/client.py]: Request to remove this commented definition.

File: hpe3parclient/client.py [hunk 5]
```diff
@@ -1880,6 +1882,29 @@ def getPorts(self):
 
         """
         response, body = self.http.get('/ports')
+        # if any of the ports are iSCSI ports and are vlan tagged
+        # then we need to get the iscsi ports and append them to
```
Comment (wdurairaj) [inline hpe3parclient/client.py]: this comment needs a change. Change will be 
```
if any of the ports are iSCSI ports and are vlan tagged (as shown by showport -iscsivlans), then
the port information is merged with the WSAPI returned port information.
```

File: test/test_HPE3ParClient_ports.py [hunk 6]
```diff
@@ -124,7 +125,7 @@ def test_cloned_iscsi_ports(self):
 
     def test_get_ports_ssh(self):
         self.printHeader('get_ports_cli')
-        self.cl.setSSHOptions('192.168.67.7','3paradm','3pardata')
+        self.cl.setSSHOptions('192.168.67.7', '3paradm', '3pardata')
```
Comment (sneharai4) [inline test/test_HPE3ParClient_ports.py]: Please replace with dummy IP and dummy credentials

**Pull request merged**